MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1325,5 +1351,209 @@ void EnggDiffractionPresenter::calcVanadiumWorkspaces(
   vanCurvesWS = ADS.retrieveWS<MatrixWorkspace>(curvesName);
 }
 
+/**
+ * Checks the plot type selected and applies the appropriate
+ * python function to apply during first bank and second bank
+ *
+ * @param outWSName title of the focused workspace
+ * @param bank the number of bank
+ */
+void EnggDiffractionPresenter::plotFocusedWorkspace(std::string outWSName,
+                                                    std::string bank) {
+  const bool plotFocusedWS = m_view->focusedOutWorkspace();
+  int plotType = m_view->currentPlotType();
+  if (plotFocusedWS) {
+    if (plotType == 0) {
+      if (bank == "1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (bank == "2")
+        m_view->plotReplacingWindow(outWSName);
+    } else if (1 == plotType) {
+      if (bank == "1")
+        m_view->plotFocusedSpectrum(outWSName);
+      if (bank == "2")
+        m_view->plotWaterfallSpectrum(outWSName);
+    } else if (2 == plotType) {
+      m_view->plotFocusedSpectrum(outWSName);
+    }
+  }
+}
+
+/**
+ * Convert the generated output files and saves them in
+ * FocusedXYE format
+ *
+ * @param inputWorkspace title of the focused workspace
+ * @param bank the number of the bank as a string
+ * @param runNo the run number as a string
+ */
+void EnggDiffractionPresenter::saveFocusedXYE(const std::string inputWorkspace,
+                                              std::string bank,
+                                              std::string runNo) {
+  auto alg = Algorithm::fromString("SaveFocusedXYE");
+  std::string fullFilename;
+  if (inputWorkspace.std::string::find("texture") != std::string::npos) {
+    fullFilename = "ENGINX_" + runNo + "_texture_" + bank + ".dat";
+  }
+  if (inputWorkspace.std::string::find("cropped") != std::string::npos) {
+    fullFilename = "ENGINX_" + runNo + "_cropped_" +
+                   boost::lexical_cast<std::string>(g_croppedCounter) + ".dat";
+    g_croppedCounter++;
+  } else {
+    fullFilename = "ENGINX_" + runNo + "_bank_" + bank + ".dat";
+  }
+  std::string saveDir;
+  try {
+    // takes to the root of directory according to the platform
+    // and appends the following string provided
+    saveDir =
+        Poco::Path(saveDir).expand("/EnginX_Mantid/User/" + runNo + "/Focus/");
```

Comment:
I think this is missing the `$HOME`, something like `expand("$HOME/Enginx_Mantid/user/")`
On linux it is writing the results under the current path from which I run Mantid and somehow it's not creating the subdirectories.

But since the behavior that they want in principle is to use `C:\EnginX_Mantid\...` as default, for windows that should be fixed, whereas for Linux the users' home looks like the best choice. You can get it with `Poco::Path::home()` and then append to that the "standard" enginx path `EnginX_Mantid/User/...`.